### PR TITLE
Feature add support to provide custom getToken function

### PR DESCRIPTION
**Pull request by jakubvilhan**
Hello,

we had an issue with current token fetching we need to use our custom fetch function with proxy agent. So I provided way to override current getToken function similar as getUserInfo.

### CODE REVIEW

**Review by WilsonLe — CHANGES_REQUESTED** [review]
I think it's better to have the getToken function returning a string, that would make it more explicit for power users of this plugin to return a valid string token.

**Comment by jakubvilhan** [comment]
Thanks for feedback! Cool that work for us also, I updated it.

File: src/callback-endpoint.ts [hunk 1]
```diff
@@ -42,24 +42,35 @@ export const createCallbackEndpoint = (
...
+            redirect_uri: redirectUri,
+            grant_type: "authorization_code",
+          }).toString(),
+        });
+        const tokenData = await tokenResponse.json();
+
+        access_token = tokenData?.access_token;
+
+        if (typeof access_token !== "string")
+          throw new Error(`No access token: ${JSON.stringify(tokenData)}`);
```
Comment (WilsonLe) [inline src/callback-endpoint.ts:72]: These 2 lines are duplicated from above, can you move it out of the if else tree?

**Review by WilsonLe — CHANGES_REQUESTED** [review]

File: src/callback-endpoint.ts (hunk 1 above)
Comment (jakubvilhan) [reply src/callback-endpoint.ts:72]: Hello, I wanted to keep logging the full json response in the default way of getting the token. Or we dont need that?

Comment (WilsonLe) [reply src/callback-endpoint.ts:72]: not in the default way, that's the point of having a custom `getToken` function

Comment (jakubvilhan) [reply src/callback-endpoint.ts]: Makes sense. Done.

**Review by WilsonLe — APPROVED** [review]

**Comment by WilsonLe** [comment]
Looks good, thank you!

**Pull request merged**